rq/worker.py

```diff
@@ -974,6 +977,12 @@ def handle_job_failure(self, job, queue, started_job_registry=None,
                                                         job_class=self.job_class, serializer=job.serializer)
                 failed_job_registry.add(job, ttl=job.failure_ttl,
                                         exc_string=exc_string, pipeline=pipeline)
+                try:
+                    pipeline.execute()
+                except Exception:
+                    # Ensure that custom exception handlers are called
+                    # even if Redis is down
+                    pass
```

Comment:
Can we remove this code block? This is not relevant for this PR and I'm not sure calling `pipeline.execute()` here is the right thing to do.

Reply:
This is required to make `test_job.py::TestJob::test_create_and_cancel_job_enqueue_dependents_in_registry` pass, because originally the job status would be stuck in the `started` state (maybe due to the addition of unconditionally enqueueing dependents below that code block in the same function)